packages/stream-api/src/webSocketClient.ts

```diff
@@ -80,28 +76,6 @@ export class WebSocketClientImpl implements WebSocketClient {
     }
   }
 
-  /**
-   * @private Used for internals or testing.
-   */
-  public async reconnect(descriptor: ConnectionDescriptor): Promise<void> {
-    await new Promise(resolve => {
-      this.timer = window.setTimeout(
-        resolve,
-        this.reconnectDelays[
-          Math.min(this.reopenAttempt, this.reconnectDelays.length - 1)
-        ]
-      );
-    });
-
-    this.reopenAttempt += 1;
-
-    try {
-      await this.connect(descriptor);
-    } catch (e) {
-      // Failed connection attempt here will be handled, and this exception can be ignored
-    }
-  }
```

Comment:
The close logic and removal of event listeners should probably be revisted now as well. In particular, we had to do some hacky things like manually emitting a close event. I wonder if this can be improved by reordering when we close the connection and remove the listeners.